src/common/utils.js

```diff
@@ -109,3 +109,20 @@ export const isJSONString = string => {
   }
   return true;
 };
+
+export const getMetadata = metadataString => {
+  const isJSONString = string => {
+    try {
+      JSON.parse(string);
+    } catch (error) {
+      return false;
+    }
+    return true;
+  };
+  const metadata =
+    metadataString && isJSONString(metadataString)
+      ? JSON.parse(metadataString)
```

Comment:
this fn can be simplified to make only one call to JSON.parse. Also let's return empty object in case of error.
```
function getMetadata(metadataStr) {
  try {
    return metadataStr === "" ? {} : JSON.parse(metadataStr);
  } catch (err) {
     console.error("metada parsing failed", err);
     return {};
  }
}
```